chainerx_cc/chainerx/routines/linalg.cc

```diff
@@ -325,4 +332,65 @@ Array PseudoInverse(const Array& a, float rcond) {
     return out;
 }
 
+std::tuple<Array, Array> Eigh(const Array& a, char uplo) {
+    CheckRankTwoArray(a);
+    CheckSquareMatrix(a);
+    CheckUplo(uplo);
+
+    Array w = Empty(Shape{a.shape()[0]}, a.dtype(), a.device());
+    Array v = Empty(a.shape(), a.dtype(), a.device());
+
+    {
+        NoBackpropModeScope scope{};
+        a.device().backend().CallKernel<SyevdKernel>(a, w, v, uplo, /*compute_v=*/true);
+        v = v.Transpose();
```

Comment:
Should we let the kernel be responsible for this transpose?

Reply:
I've moved `v = v.Transpose()` to the kernels.
It's done with
`device.backend().CallKernel<CopyKernel>(v.Transpose(), v);`